oauth2-server/src/main/java/com/clouway/oauth2/ClientAuthorizationActivity.java

```diff
@@ -41,7 +41,7 @@ public Response execute(String identityId, Request request) {
     String codeVerifierMethod = request.param("code_challenge_method") == null ? "" : request.param("code_challenge_method");
     CodeChallenge codeChallenge = new CodeChallenge(codeChallengeValue, codeVerifierMethod);
 
-    Map<String, String> params = new Params().parse(request, "response_type", "client_id", "redirect_uri", "state", "scope", "code_challenge", "code_challenge_method");
+    Map<String, String> params = new Params().parse(request, "response_type", "redirect_uri", "state", "scope", "code_challenge", "code_challenge_method");
```

Comment:
I can agree with this change, but I'm not sure that this is the best solution. 
The other solution could 
```
Optional<Identity> possibleIdentity = identityFinder.findIdentity(authorization.identityId, GrantType.AUTHORIZATION_CODE, instant, authorization.params, **authorization.clientId**);
```
adding the clientId to the identityFinder.findIdentity(...) method

the main purpose is the clientId to be provided to the identityFinder.findIdentity(...), so we can fetch the correct customer in case that the client is not internal and is owned from a single customer.

Reply:
I am agree with your suggestion, but doesnt it look strange for the method 
``` findIdentity(.. clientId: String) ``` to pass oauth client id that is used for authorization, when you looking for identity ?

Reply:
If this change adds clientId to the authorization object it sounds reasonable to me.

Reply:
No this change doesnt add clientId to the authorization object, the authorization object already has client id as a field

Reply:
This answer doesn't give me any additional information. 

This is what I see in tests: 
```java
 oneOf(clientAuthorizationRepository).authorize(new AuthorizationRequest(anyExistingClient, "user1", "code", Collections.singleton("abc"), emptyCodeChallenge, ImmutableMap.of("customerIndex", "::customerIndex::", "client_id", "::client_id::")));
      will(returnValue(Optional.of(new Authorization("code", "::client_id::", "identityId", "1234", Collections.<String>emptySet(), Collections.singleton("http://example.com/callback"), Collections.singletonMap("customer", "::customerIndex::")))));
```
e.g 

the clientAuthorizationRepository.authorize is called with clientID and returns Authorization object with additional custom params. Note that this "customerId" is moved back to the library which doesn't sound reasonable as it couples it with internal of an app.

Reply:
This PR is for that to provide clientId  to identity finder. With that client the finder will be able to select the "customer" of users. This is the purpose of this change.

Reply:
You mean that it will return an identity with appropriate custom claims associated with it ?

Reply:
Yes